google-cloud-firestore/samples/CONTRIBUTING.md

```diff
@@ -0,0 +1,289 @@
+   cd google-cloud-ruby
+   git remote add upstream git@github.com:googleapis/google-cloud-ruby.git
+   ```
+
+1. If your fork and clone are not brand new, get the latest changes from `upstream`:
+
+   ```bash
+   git checkout main
+   git pull upstream main
+   ```
+
+1. Create a new topic branch off of the `main` branch:
+
+   ```bash
+   git checkout -b <topic-branch>
+   ```
+
+1. Change to the library's sub-directory in the repo:
+
+   ```sh
+   $ cd google-cloud-firestore
+   ```
+
+1. Install (or update) the library dependencies:
+
+   ```sh
+   $ bundle update
+   ```
+
+1. Change to the library samples sub-directory:
+
+   ```sh
+   $ cd google-cloud-firestore/samples
+   ```
+
+1. Install (or update) the library samples dependencies:
+
+   ```sh
+   $ bundle update
+   ```
+
+### Acceptance tests setup
+
+To set up your acceptance test credentials:
+
+1. If needed, create a Google Cloud project. In the Google Cloud Console, on the project selector page, select or create
+   a project.
+
+1. Ensure that billing is enabled for your project.
+
+1. Ensure that the Firestore API is enabled for your project. Note that if you have already enabled the Datastore API
+   for your project, you will need to use a different project for Firestore.
+
+1. Follow the instructions for [Creating a Service Account](AUTHENTICATION.md#creating-a-service-account) in
+   `AUTHENTICATION.md`, including downloading and securely storing a JSON key file. 
+
+1. Set the `GCLOUD_TEST_KEYFILE` environment variable to the path of the JSON key file that you downloaded in the
+   previous step:
+
+   ``` sh
+   $ export GCLOUD_TEST_KEYFILE=/path/to/keyfile.json
+   ```
+
+   If you are already using the `GCLOUD_TEST_KEYFILE` environment variable, and wish to test the samples with a
+   different key file, you may set the `FIRESTORE_TEST_KEYFILE` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_TEST_KEYFILE=/path/to/keyfile.json
+   ```
+
+1. Set the `GCLOUD_TEST_PROJECT` environment variable to your Google Cloud project ID:
+
+   ``` sh
+   $ export GCLOUD_TEST_PROJECT=my-project-id
+   ```
+
+   If you are already using the `GCLOUD_TEST_PROJECT` environment variable, and wish to test the samples with a
+   different project, you may set the `FIRESTORE_TEST_PROJECT` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_TEST_PROJECT=my-project-id
+   ```
+
+## Run CI
+
+You are now ready to run local CI checks for the samples, which you should do **before** you make any changes. Doing so
+ensures that everything is OK with your local environment and the latest dependency versions. You don't want any
+surprises later.
+
+If you haven't already done so, change to the library's sub-directory in the repo:
+
+```sh
+$ cd google-cloud-firestore
+```
+
+Note: Do not change to the `google-cloud-firestore/samples` directory. If you are already in the samples directory,
+move up to the library directory:
+
+```sh
+$ cd ../google-cloud-firestore
```

Comment:
This applies if you're in a separate library directory, not if you're in the samples directory. I think we can just omit the example.

Reply:
If you're in the `samples` directory, you need to move up a level to the `google-cloud-firestore` directory in order to run `rake samples`. This is one of the more confusing things about working with samples imo.

Reply:
There's a bit more to consider: 

Running `rake samples` in the `google-cloud-firestore` directory invokes `bundle update` every time, so it's a slow way to run a single test with focus repeatedly during development. For that, `rake test` in the `samples directory is better. 

On the other hand, there is no `rake rubocop` task in the `samples` directory, although running `bundle exec rubocop` *should* work. (I vaguely remember having an issue with it in the past, I think because it picks up configuration from the parent directory `.rubocop` file, but I can't remember exactly what situation produces the problem. Appears to work for me now.)

Reply:
There's nothing stopping us from defining a `rake samples` task in `samples/Rakefile`. Or a `rake rubocop` task in that same Rakefile for that matter.

Another option is to switch to Toys, which can gracefully handle being run from a subdirectory. So we can define e.g. `toys samples` or `toys test --samples` at the `google-cloud-firestore` level, and it will still work in the `samples` directory. And similar for `toys rubocop`.

Regarding `bundle update`, is that actually necessary? Could we just change the task to omit it? (Or maybe do a `bundle check` first to see if it's actually needed?)

Reply:
These are good suggestions. For now, I'll simply change the instructions to show `bundle exec rubocop` and `bundle exec rake test` in the `samples` directory. Both currently work in google-cloud-firestore, and running checks in the `samples` directory is the nicest workflow for the developer. We can separately investigate whether `bundle update` is necessary. (I asked about it when we first introduced samples to this repo, and iirc it was something to do with automated CI, but not a firm requirement.)

Reply:
Updated, PTAL!